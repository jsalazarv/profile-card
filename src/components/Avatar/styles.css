.avatar {
    @apply flex;
    @apply items-center;
    @apply justify-center;
    @apply overflow-auto;
}
