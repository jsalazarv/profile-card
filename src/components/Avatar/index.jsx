import './styles.css';
import { useMemo } from 'react';

export const Avatar = ({
  className = '',
  children,
  size = 'small',
  type = 'circle',
  color = 'bg-primary',
}) => {
  const sizes = {
    small: 'w-6 h-6',
    medium: 'w-16 h-16',
    large: 'w-32 h-32',
  };

  const types = {
    circle: 'rounded-full',
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };

  const avatarSize = useMemo(() => sizes[size], [size]);
  const avatarType = useMemo(() => types[type], [type]);

  return (
    <div
      className={`overflow-auto ${avatarSize} ${avatarType} ${color} ${className}`}>
      {children}
    </div>
  );
};
