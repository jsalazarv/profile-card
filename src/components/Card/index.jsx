import './styles.css';
import { useMemo } from 'react';

export const Card = ({
  className = '',
  children,
  type = 'tile',
  outlined = '',
}) => {
  const types = {
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };
  const outline = 'border-solid border';
  const cardType = useMemo(() => types[type], [type]);

  return (
    <div className={`card ${cardType} ${outlined && outline} ${className}`}>
      {children}
    </div>
  );
};
