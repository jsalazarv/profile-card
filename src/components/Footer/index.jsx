import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import './styles.css';

export const Footer = ({ phrase = 'Designed with' }) => {
  return (
    <>
      <footer className="footer">
        <p>
          {phrase}
          <span>
            <FontAwesomeIcon icon={faHeart} />
          </span>
        </p>
      </footer>
    </>
  );
};
