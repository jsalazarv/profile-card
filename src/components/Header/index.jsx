import './styles.css';
import { ThemeToggle } from '../ThemeToggle/index.jsx';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

export const Header = ({}) => {
  return (
    <header className="header">
      <nav className="header__navbar" aria-label="Top">
        <div className="navbar__content">
          <div className="logo__container">
            <a href="https://jsalazarv.dev/" target="_blank">
              <h2>[Js]</h2>
            </a>
          </div>
          <div className="items__container">
            <a href="https://github.com/jsalazarv/profile-card" target="_blank">
              <FontAwesomeIcon
                className="text-2xl border-black dark:text-amber-500 dark:text-3xl transition-all duration-1000"
                icon={faGithub}
              />
            </a>
            <ThemeToggle />
          </div>
        </div>
      </nav>
    </header>
  );
};
