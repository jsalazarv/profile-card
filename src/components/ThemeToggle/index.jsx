import './styles.css';
import useDarkMode from '../../hooks/useDarkMode.js';

export const ThemeToggle = ({ className = '' }) => {
  const [colorTheme, setTheme] = useDarkMode();

  const toggleTheme = () => {
    setTheme(colorTheme);
  };

  return (
    <input
      className={`toggle ${className}`}
      type="checkbox"
      onClick={toggleTheme}
      aria-label="toggleTheme"
    />
  );
};
