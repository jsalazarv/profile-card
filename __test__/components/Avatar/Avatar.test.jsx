import { render } from '@testing-library/react';
import { Avatar } from '../../../src/components/Avatar';

describe('[Avatar Component]', () => {
  describe('[Snapshot]', () => {
    test('Should match snapshot', () => {
      const avatarType = 'circle';
      const avatarSize = 'large';
      const { container } = render(
        <Avatar type={avatarType} size={avatarSize} />,
      );

      expect(container).toMatchSnapshot();
    });

    test('Properties configured on the component should only be available options', () => {
      const sizeOptions = ['small', 'medium', 'large'];
      const typeOptions = ['circle', 'rounded', 'tile'];
      const avatarSize = 'large';
      const avatarType = 'circle';
      const { container } = render(
        <Avatar type={avatarType} size={avatarSize}>
          <img
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQGMsJNRv8en8a8mnoT_W9D7CjqV55SAPTurg&usqp=CAU"
            alt=""
          />
        </Avatar>,
      );

      expect(new Set(sizeOptions)).toContain(avatarSize);
      expect(new Set(typeOptions)).toContain(avatarType);
      expect(container).toMatchSnapshot();
    });

    test('Should process child properties (img) correctly', () => {
      const avatarType = 'circle';
      const avatarSize = 'large';
      const { container } = render(
        <Avatar type={avatarType} size={avatarSize}>
          <img
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQGMsJNRv8en8a8mnoT_W9D7CjqV55SAPTurg&usqp=CAU"
            alt=""
          />
        </Avatar>,
      );

      expect(container).toMatchSnapshot();
    });

    test('Should process child properties (text) correctly', () => {
      const avatarType = 'circle';
      const avatarSize = 'large';
      const { container } = render(
        <Avatar type={avatarType} size={avatarSize}>
          <span>JS</span>
        </Avatar>,
      );

      expect(container).toMatchSnapshot();
    });

    test('Should process child properties (size, type) correctly', () => {
      const avatarType = 'rounded';
      const avatarSize = 'small';
      const { container } = render(
        <Avatar type={avatarType} size={avatarSize}>
          <img
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQGMsJNRv8en8a8mnoT_W9D7CjqV55SAPTurg&usqp=CAU"
            alt=""
          />
        </Avatar>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
