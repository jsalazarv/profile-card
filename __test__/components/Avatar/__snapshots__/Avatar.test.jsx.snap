// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Avatar Component] [Snapshot] It should contain the class [rounded-full] if the configured prop is [circle] 1`] = `
<div>
  <div
    class="avatar w-6 h-6 rounded-full transparent "
  />
</div>
`;

exports[`[Avatar Component] [Snapshot] It should contain the class [rounded-lg] if the configured prop is [rounded] 1`] = `
<div>
  <div
    class="avatar w-6 h-6 rounded-lg transparent "
  />
</div>
`;

exports[`[Avatar Component] [Snapshot] It should contain the class [rounded-none] if the configured prop is [tile] 1`] = `
<div>
  <div
    class="avatar w-6 h-6 rounded-none transparent "
  />
</div>
`;

exports[`[Avatar Component] [Snapshot] It should represent the default values 1`] = `
<div>
  <div
    class="avatar w-6 h-6 rounded-full transparent "
  />
</div>
`;
